src/openfermion/measurements/_hamiltonian_stabilizers_sets.py

```diff
@@ -0,0 +1,126 @@
+            has been split into.
+    Return:
+        left_paulis(QubitOperator): Only if there are left strings.
+            If no missing Paulis, returns an empyt QubitOperator.
+    """
+    # This functions uses python object sets to compare missing tuples.
+    # Hamiltonian keys as a set.
+    ham_set_form = set(hamiltonian.terms.keys())
+
+    # Hamiltonian subsets as a single set object.
+    aux_set = set()
+    for sb in subsets:
+        aux_set.update(set(sb.terms.keys()))
+
+    # Find any missing tuple in the Hamiltonian set compared to
+    # all subsets.
+    left_set = ham_set_form - aux_set
+
+    # Save missing Pauli strings in QubitOperator
+    # If no missing Paulis, returns 0
+    left_paulis = QubitOperator()
+    for p in left_set:
+        left_paulis += QubitOperator(p, hamiltonian.terms[p])
+    return left_paulis
+
+
+def get_hamiltonian_subsets(hamiltonian, stabilizers):
+    """
+    Create subsets of Hamiltonian.
+
+    This function splits a Hamiltonian into subsets of Pauli strings that do
+    not overlap with stabilizers.
+    Pauli strings might be repeated between subsets if they do not share any
+    qubit with the stabilizer.
+    The Pauli strings that overlap with all stabilizers will be stored with a
+    separate set.
+    It can be used to perform SV and not disregarding all data if only a single
+    stabilizers signals an error.
+
+    Args:
+        hamiltonian(QubitOperator): Hamiltonian to be split.
+        stabilizers(list or QubitOperator): List of stabilizers as
+            QubitOperators.
+    Return:
+        hamiltonian_subsets(list): List of QubitOperator which are a subset of
+            the Hamiltonian.
+            Each QubitOperator does not overlap with the stabilizer at the same
+            position in the list.
+        rest_paulis(QubitOperator): Operator that contains the Pauli
+            strings that overlap with all the stabilizers.
+            If there are no paulis left an empty QubitOperator is returned.
+    Raises:
+        TypeError: Input hamiltonian must be QubitOperator.
+        TypeError: Input stabilizer_list must be QubitOperator or list.
+    """
+    if not isinstance(hamiltonian, QubitOperator):
+        raise TypeError('Input terms must be QubitOperator.')
+    if not isinstance(stabilizers, (QubitOperator, list,
+                                    tuple, numpy.ndarray)):
+        raise TypeError('Input stabilizers must be QubitOperator or list.')
+
+    hamiltonian_subsets = []
+    for stab in list(stabilizers):
+        # Initialize subset
+        ham_subset = QubitOperator()
+        for pauli in hamiltonian:
+            if _has_overlapping_indices(pauli, stab) is False:
+                ham_subset += pauli
+
+        # Adds the subset to the list.
+        hamiltonian_subsets.append(ham_subset)
+
+    # Check if there are missing strings
+    rest_paulis = _check_missing_paulis(hamiltonian, hamiltonian_subsets)
```

Comment:
Instead of executing this function to construct the QubitOperator with the remaining Pauli strings, construct it in the following more efficient way: At the beginning of this function, make a copy of the original Hamiltonian. Then, whenever you add a Pauli string to a subset, delete it from this copy. At the end, this copy will be the thing you want. Also, rename `rest_paulis` to `remaining_paulis`.

Reply:
Hi Kevin,
Thanks for taking the time to look into this.

Indeed, it was very dumb to check remaining operators in a different function. Thanks for you help there.